lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -708,6 +709,10 @@ private function getPreAlterTableAlterPrimaryKeySQL(TableDiff $diff, Index $inde
 
         // Dropping primary keys requires to unset autoincrement attribute on the particular column first.
         foreach ($index->getColumns() as $columnName) {
+            if(isset($diff->addedColumns[$columnName]) && $diff->addedColumns[$columnName] instanceof Column) {
+                continue;
+            }
```

Comment:
I think using the following check is more convenient and safer concerning the following `getColumn()` call:

```php
if (! $diff->fromTable->hasColumn($columnName)) {
    continue;
}
```